zip.ml

```diff
@@ -210,7 +210,7 @@ let read_cd filename ic cd_entries cd_offset cd_bound =
            } :: !e
     done;
     assert((cd_bound = (LargeFile.pos_in ic)) &&
-           (cd_entries = 65535 || !entrycnt = cd_entries));
+           (cd_entries = 65535 || cd_entries = !entrycnt land 0xffff || !entrycnt = cd_entries));
     List.rev !e
```

Comment:
It seems that `cd_entries` is always between 0 and 0xFFFF, because it's represented by 2 bytes in the ZIP file.  So, the old `!entrycnt = cd_entries` test is redundant with the new test, `cd_entries = !entrycnt land 0xffff`.  It doesn't hurt either.

Reply:
Yep, you're correct.